### PR TITLE
Update CI to cuda13

### PR DESCRIPTION
**Pull request by coreyjadams**
# PhysicsNeMo Pull Request

This PR updates our CI from cuda12 to cuda13.

It also fixes a bug in how we check yaml files for license headers.  So it looks a lot bigger than it is, because I also fixed all those license headers.

Additionally, the nightly container and pre-commit workflows are deleted:
- nightly container was not used.
- pre-commit is superceded by the `app` version of the same tool, which is faster and more efficient.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: .github/ci-requirements.txt [hunk 1]
```diff
@@ -61,12 +61,21 @@ pyarrow==24.0.0
...
 # unclaimed on PyPI (the team publishes there as `pyg_library`), so a
-# plain `==0.6.0` pin could match a squatted package.
-torch_scatter==2.1.2+pt211cu128
-torch_sparse==0.6.18+pt211cu128
-torch_cluster==1.6.3+pt211cu128
-pyg_lib==0.6.0+pt211cu128
+# plain `==0.8.0` pin could match a squatted package.
+torch_scatter==2.1.2+pt212cu130
+torch_sparse==0.6.18+pt212cu130
+pyg_lib==0.8.0+pt212cu130
```
[the remark this answers is not in the document]
Comment (coreyjadams) [reply .github/ci-requirements.txt:78]: This is KNOWN.  We need to regenerate the lock file before this deploys.  Thanks captain obvious for reading the docs.

File: .github/workflows/github-nightly-uv.yml [hunk 2]
```diff
@@ -79,14 +79,14 @@ env:
   # Keep CONTAINER_ID in sync with the `image:` tag below.
   PYTHON_VERSION: "3.12"
```
Comment (peterdsharpe) [inline .github/workflows/github-nightly-uv.yml]: Maybe bump this (and similar instances) to 3.13, but this is totally optional

File: .github/workflows/github-nightly-uv.yml [hunk 3]
```diff
@@ -79,14 +79,14 @@ env:
...
   UV_VERSION: "0.11.7"
```
Comment (peterdsharpe) [inline .github/workflows/github-nightly-uv.yml]: Might be worth updating this while we're at it, but again, totally optional

**Review by peterdsharpe — APPROVED** [review]
Great calls all around. This bundles a bunch of small smart changes.

File: .github/workflows/github-nightly-uv.yml (hunk 2 above)
Comment (coreyjadams) [reply .github/workflows/github-nightly-uv.yml]: Done.

File: .github/workflows/github-nightly-uv.yml (hunk 3 above)
Comment (coreyjadams) [reply .github/workflows/github-nightly-uv.yml]: Done.

File: .github/actions/setup-uv-env/action.yml [hunk 4]
```diff
@@ -201,11 +242,16 @@ runs:
...
+        # setup.py imports torch at build time; disabling isolation for
+        # just that package lets the build see the synced .venv's torch
+        # and setuptools instead of failing in an empty build env.
         uv pip install --python .venv/bin/python \
           --reinstall-package torch_scatter \
           --reinstall-package torch_sparse \
           --reinstall-package torch_cluster \
           --reinstall-package pyg_lib \
+          --no-build-isolation-package torch_cluster \
           --constraint .github/ci-requirements.lock \
```
Comment (ktangsali) [inline .github/actions/setup-uv-env/action.yml:264]: Is Natten required here as well? I remember including it for the docker builds

File: .github/regen-ci-deps-lock.sh [hunk 5]
```diff
@@ -94,20 +97,25 @@ for e in "${extras[@]}"; do
...
 echo "::: install CI-only deps + PyG CUDA wheels ..."
-UV_LINK_MODE=copy uv pip install --python .venv/bin/python \
+UV_LINK_MODE=copy FORCE_CUDA=1 TORCH_CUDA_ARCH_LIST=9.0 \
+  uv pip install --python .venv/bin/python \
   --reinstall-package torch_scatter \
   --reinstall-package torch_sparse \
   --reinstall-package torch_cluster \
   --reinstall-package pyg_lib \
+  --no-build-isolation-package torch_cluster \
   -r .github/ci-requirements.txt
```
Comment (ktangsali) [inline .github/regen-ci-deps-lock.sh:129]: same here about natten. Please check

**Review by ktangsali — APPROVED** [review]
Looks mostly good, @coreyjadams did you mention you changed something on the license testing front? I don't see those diffs here.

**Comment by ktangsali** [comment]
> Looks mostly good, @coreyjadams did you mention you changed something on the license testing front? I don't see those diffs here.

Was it this change? 

      files: \.(py|yaml|yml|ci|release)$|Dockerfile$

**Comment by coreyjadams** [comment]
Yes: it was exactly the change you mentioned.  The license check was only looking for `.yaml` files, and most of our github files are `.yml`, and the difference is `load bearing`.

After I added `.yml` to that line so it searched them too, it picked up many failures that we had been allowing (only half were mine, I promise!) so I fixed the ones it finds.

File: .github/actions/setup-uv-env/action.yml (hunk 4 above)
Comment (coreyjadams) [reply .github/actions/setup-uv-env/action.yml:264]: No, natten is not required here.  Yes, we _do_ include natten in the builds, but this section is a funky song and dance we do just for the PyG eco system.

PyG ships `sdist` on pypi only, and because we resolve that and build it on the CPU nodes that is a cpu only build that arrives here.  On the GPU nodes we reinstall that and force CUDA.  For `torch_cluster`, in this line, we let it see the whole python env too so it can import torch at build time and resolve cuda deps through torch.

tl;dr: no, natten not needed because we include it elsewhere.

File: .github/regen-ci-deps-lock.sh (hunk 5 above)
Comment (coreyjadams) [reply .github/regen-ci-deps-lock.sh:129]: Same answer :)

**Comment by coreyjadams** [comment]
/ok to test [5cbd2de](https://github.com/NVIDIA/physicsnemo/pull/1950/commits/5cbd2de4c6ae6b4b07c16c03f4ddf3add95b6edc)

**Comment by coreyjadams** [comment]
/ok to test [af01ae7](https://github.com/NVIDIA/physicsnemo/pull/1950/commits/af01ae7b5ca14be0e4ef540c389ce651f962de5f)

**Comment by coreyjadams** [comment]
/ok to test [de3fa22](https://github.com/NVIDIA/physicsnemo/pull/1950/commits/de3fa22b6088fc722842a9138d879029ffc49109)

**Comment by peterdsharpe** [comment]
Recommend bumping the single-run CI for PRs to 3.13 or 3.14? Optional, but could go here

**Comment by coreyjadams** [comment]
Closing this PR; I am rebasing it on to the fresh `main` since we have CI updates there and it's cleaner to rebase and reopen.
